amivapi/ldap.py

```diff
@@ -129,22 +133,15 @@ def _filter_data(data):
     res['gender'] = \
         u"male" if int(data['swissEduPersonGender']) == 1 else u"female"
 
-    if 'D-ITET' in data['ou']:
-        res['department'] = u"itet"
-    elif 'D-MAVT' in data['ou']:
-        res['department'] = u"mavt"
-    else:
-        res['department'] = u"other"
-
-    # ou contains all 'organization units'. This contains fields of study.
-    # Check if it contains any field of study assigned to us
-    ou_list = current_app.config['LDAP_MEMBER_OU_LIST']
-    is_member = bool(set(data['ou']).intersection(set(ou_list)))
-
-    if ('VSETH Mitglied' in data['ou']) and is_member:
-        res['membership'] = u"regular"
-    else:
-        res['membership'] = u"none"
+    # The departmentNumber field contains certain phrases for each department
+    department_map = current_app.config['LDAP_DEPARTMENT_MAP'].items()
+    department = (dept for phrase, dept in department_map
+                  if phrase in data['departmentNumber'][0])
+    res['department'] = next(department, None)  # None if no match
+
+    # Membership: One of our departments and VSETH member
+    is_member = res['department'] and ('VSETH Mitglied' in data['ou'])
```

Comment:
I think pytest would complain here to make it more explicit and write `res['department'] is not None and ...`